### PR TITLE
feat(cli): cli as k8s deployment for debugging

### PR DESCRIPTION
**Pull request by sakoush**
**What this PR does / why we need it**:

This PR creates a docker image `seldonio/seldon-cli` with Seldon CLI prepackaged  in it. Users can then deploy this image in Kubernetes alongside Core 2 installation in the same namespace.

This PR also factors out  kafka config code into a top level component so that it can be used by the cli and especially in a Kubernetes deployment where it can have access to `seldon-kafka` ConfigMap for consumer config.

We have disabled also Seldon CLI control plane operations (e.g. `load` and `unload`) by default as they can cause inconsistencies in Kubernetes installation. They can be re-enabled with the flag `--force`. 

Docs have been also adjust and perhaps it would be a good entrypoint to review the changes of this PR. Link to rendered docs [here](https://app.gitbook.com/o/rOXFWV2k010iw8QdOyaH/s/n8ONqKsfp9k7WHp5OWSM/~/revisions/KwMNjKCFAuo2vprMDjNr/cli).

**Which issue(s) this PR fixes**:

Fixes Infra-1269 (internal)

**Special notes for your reviewer**:

- Unit tests have been added for this change.
- CI changes added to include tests and builds of the new top level component.

### CODE REVIEW

File: operator/config/cli/seldon-cli-remote.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+
+
+# This script is used to run commands on the seldon-cli pod in the seldon-mesh namespace
+# Usage: ./seldon-cli-remote.sh <command>
+# command: the command to run on the seldon-cli pod in the format: `seldon` <args> 
+
+N="${NAMESPACE:-seldon-mesh}"
+
+kubectl exec seldon-cli -n ${N} -- $*
```
Comment (driev) [inline operator/config/cli/seldon-cli-remote.sh:19]: newline here and in the file below.

**Review by driev — APPROVED** [review]
Looks good - left a few small comments.

File: components/kafka/Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
+GO_LDFLAGS := -w $(patsubst %,-X %, $(GO_BUILD_VARS))
+
+.PHONY: test
+test:
+	go test ./pkg/... -coverprofile cover.out
+
+.GOLANGCILINT_VERSION := v1.57.2
```
[the remark this answers is not in the document]
Comment (sakoush) [reply components/kafka/Makefile:7]: Lets leave it to a follow up PR as the version 1.57.2 is also used elsewhere and it is consistent.

File: docs-gb/cli/README.md [hunk 3]
```diff
@@ -4,37 +4,128 @@ description: Learn more about using Seldon CLI commands
...
+
+Seldon CLI can be deployed in the same namespace along side a Core 2 deployment, which would allow users to have access to the different CLI commands with minimal setup (i.e. envars set appropriately). 
+
+### Deployment
+
+`seldonio/seldon-cli` Docker image has prepackaged Seldon CLI suitable for deployment in a Kubernetes cluster. Consider the following Kubernetes Deployment manifest: 
+
+```yaml
+apiVersion: v1
+kind: Pod
```
[the remark this answers is not in the document]
Comment (sakoush) [reply docs-gb/cli/README.md:30]: I think this example works as is as I have tested it, perhaps we can convert it later to a Deployment?
I dont think that this is really a long running deployment as well, it should just be for the duration of a debugging session perhaps.

[the remark this answers is not in the document]
Comment (driev) [reply docs-gb/cli/README.md:30]: That's fine... the documentation mentions deployment, so change that for conistency.

**Pull request merged**